### ISSUE TITLE
[Bug] Docker build fails (boringssl)

### ISSUE THREAD

**Issue by prototux** [iv-org/invidious#2310]
**Describe the bug**
The docker build fails at boringssl, it may be a alpine bug (?) i don't know when it started to break exactly since i only update invidious when i have time to (mono-user private instance)

**Steps to Reproduce**
run `docker-compose build`

**Logs**
```
Step 6/37 : RUN cd boringssl && abuild -F -r && cd ..
 ---> Running in f5cbb16b312c
>>> boringssl: Building src/boringssl 1.1.0-r0 (using abuild 3.8.0_rc4-r0) started Sun, 08 Aug 2021 15:17:02 +0000
>>> boringssl: Checking sanity of /src/boringssl/APKBUILD...
>>> boringssl: Analyzing dependencies...
>>> boringssl: Installing for build: build-base !openssl-libs-static cmake git go perl
WARNING: Ignoring /root/packages//src: No such file or directory
fetch https://dl-cdn.alpinelinux.org/alpine/edge/main/x86_64/APKINDEX.tar.gz
fetch https://dl-cdn.alpinelinux.org/alpine/edge/community/x86_64/APKINDEX.tar.gz
(1/1) Installing .makedepends-boringssl (20210808.151703)
OK: 636 MiB in 68 packages
>>> boringssl: Cleaning up srcdir
>>> boringssl: Cleaning up pkgdir
>>> boringssl: Fetching 251b516.tar.gz::https://github.com/google/boringssl/tarball/251b516
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   165  100   165    0     0   3316      0 --:--:-- --:--:-- --:--:--  3367
100 25.4M    0 25.4M    0     0  8374k      0 --:--:--  0:00:03 --:--:-- 9431k
>>> boringssl: Fetching 251b516.tar.gz::https://github.com/google/boringssl/tarball/251b516
>>> boringssl: Checking sha512sums...
251b516.tar.gz: OK
>>> boringssl: Unpacking /var/cache/distfiles/251b516.tar.gz...
-- The C compiler identification is GNU 10.3.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - failed
-- Check for working C compiler: /usr/bin/gcc
-- Check for working C compiler: /usr/bin/gcc - broken
CMake Error at /usr/share/cmake/Modules/CMakeTestCCompiler.cmake:69 (message):
  The C compiler

    "/usr/bin/gcc"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: /src/boringssl/src/google-boringssl-251b516/CMakeFiles/CMakeTmp
    
    Run Build Command(s):/usr/bin/make -f Makefile cmTC_24840/fast && /usr/bin/make  -f CMakeFiles/cmTC_24840.dir/build.make CMakeFiles/cmTC_24840.dir/build
    make: /usr/bin/make: Operation not permitted
    make: *** [Makefile:127: cmTC_24840/fast] Error 127
    
    

  

  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt:14 (enable_language)


-- Configuring incomplete, errors occurred!
See also "/src/boringssl/src/google-boringssl-251b516/CMakeFiles/CMakeOutput.log".
See also "/src/boringssl/src/google-boringssl-251b516/CMakeFiles/CMakeError.log".
>>> ERROR: boringssl: build failed
>>> boringssl: Uninstalling dependencies...
(1/1) Purging .makedepends-boringssl (20210808.151703)
OK: 636 MiB in 67 packages
```

**Additional context**
the docker-compose file i use is an slightly edited version of the one in the repo, the structure of my setup is `/srv/invidious/docker-compose.yml` + invidious' repo cloned in `/srv/invidious/invidious`. the build context is (of course) modified to be `/srv/invidious/invidious`, so it shouldn't be the issue, especially since it ran without issues since about a year ago

**Comment by SamantazFox** [comment]
[Our CI passes nicely](https://github.com/iv-org/invidious/runs/3260698884#step:7:441):
```
#31 5.279 -- Detecting C compiler ABI info
#31 5.339 -- Detecting C compiler ABI info - done
#31 5.353 -- Check for working C compiler: /usr/bin/gcc - skipped
#31 5.354 -- Detecting C compile features
#31 5.355 -- Detecting C compile features - done
```

It seems that in your case, CMake can't detect the C compiler ABI infos. Try to compare our CI output with yours.

You said you run a modified `docker-compose.yml` file; are you using the same `dockerfile`s as us?

**Comment by prototux** [comment]
The dockerfile is the same, after some research i've came across https://gitlab.alpinelinux.org/alpine/aports/-/issues/12321 which may explain it.

I'm upgrading docker/libseccomp and see if it fixes the issue, as they seem to be the culprit here.

**Comment by prototux** [comment]
It indeed fix the issue (upgrading from docker 19.03.6 to 20.10.8 and libseccomp from 2.4.2 to 2.5.1), so alpine edge isn't compatible with older versions of docker and libseccomp.

Sorry for the noise, closing the issue :)

**Comment by SamantazFox** [comment]
Glad you managed to find what was your issue :)
